### PR TITLE
Deprecate `prefer-exponentiation-operator` rule

### LINKED ISSUE #483: Docs: direct users towards core `prefer-exponentiation-operator` rule
eslint's `6.7.0` release has a [`prefer-exponentiation-operator`](https://eslint.org/docs/rules/prefer-exponentiation-operator) rule. 

I don't think it's worthwhile removing the rule entirely from the plugin (users might be stuck on older version of eslint), but a good first step towards deprecation might be to start directing users to the core rule on the docs.

Thoughts?

### PR DESCRIPTION
**Pull request by fisker**
Deprecate `unicorn/prefer-exponentiation-operator` rule, replace it with ESLint official rule [`prefer-exponentiation-operator`](https://eslint.org/docs/rules/prefer-exponentiation-operator)

I ran `prefer-exponentiation-operator` over our test cases, little difference:

1. spaces

```js
Math.pow(1, 2);

1 ** 2; // unicorn/prefer-exponentiation-operator
1**2; // prefer-exponentiation-operator
```

2. exponent parenthesised

```js
Math.pow(1, +2);

1 ** +2; // unicorn/prefer-exponentiation-operator
1**(+2); // prefer-exponentiation-operator
```

fixes #483

### CODE REVIEW

**Comment by fisker** [comment]
//cc @gurpreetatwal @sindresorhus

File: test/package.js [hunk 1]
```diff
@@ -77,14 +81,17 @@ test('Every rule is defined in readme.md usage and list of rules in alphabetical
...
+
+	for (const name of availableRules) {
 		t.truthy(usageRules[`unicorn/${name}`], `'${name}' is not described in the readme.md ## Usage`);
 		t.truthy(rules.includes(name), `'${name}' is not described in the readme.md ## Rules`);
 	}
 
-	t.is(Object.keys(usageRules).length - ignoredRules.length, ruleFiles.length, 'There are more rules in readme.md ## Usage than rule files.');
-	t.is(Object.keys(rules).length, ruleFiles.length, 'There are more rules in readme.md ## Rules than rule files.');
+	t.is(Object.keys(usageRules).length - ignoredRules.length, availableRules.length, 'There are more rules in readme.md ## Usage than rule files.');
+	t.is(Object.keys(rules).length, availableRules.length, 'There are more rules in readme.md ## Rules than rule files.');
```
Comment (gurpreetatwal) [inline test/package.js]: Would it be worth it to add a check that ensures deprecated rules are described in the readme?

Comment (fisker) [reply test/package.js]: I don't think it's really nessary.

File: readme.md [hunk 2]
```diff
@@ -137,6 +135,11 @@ Configure it in `package.json`.
 - [regex-shorthand](docs/rules/regex-shorthand.md) - Enforce the use of regex shorthands to improve readability. *(fixable)*
 - [throw-new-error](docs/rules/throw-new-error.md) - Require `new` when throwing an error. *(fixable)*
 
+## Deprecated Rules
+
+|Deprecated rule|Replaced by|Rule Details|
+|:--|:--|:--|
+|[unicorn/prefer-exponentiation-operator](docs/rules/prefer-exponentiation-operator.md)|[prefer-exponentiation-operator](https://eslint.org/docs/rules/prefer-exponentiation-operator)|Prefer the exponentiation operator over `Math.pow()`|
```
Comment (sindresorhus) [inline readme.md]: I don't think we should use a table for this. Markdown tables are annoying to edit and comes with little benefit here. Not every deprecated rule will have a replacement either. Better to just use a plain list.

Comment (fisker) [reply readme.md]: done.

**Pull request merged**